core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/AbstractReplaceInvalidHyperLinkInDescription.java

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+ * Copyright (c) 2023 THALES GLOBAL SERVICES.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.platform.sirius.sirius.validation.ddiagram;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.sirius.viewpoint.DRepresentationDescriptor;
+import org.eclipse.swt.widgets.Display;
+import org.polarsys.capella.core.data.capellacore.CapellaElement;
+import org.polarsys.capella.core.diagram.helpers.DiagramHelper;
+import org.polarsys.capella.core.platform.sirius.sirius.validation.parser.helper.DescriptionParserHelper;
+import org.polarsys.kitalpha.richtext.widget.tools.dialogs.MDELinkDialog;
+import org.polarsys.kitalpha.richtext.widget.tools.manager.LinkManager;
+import org.polarsys.kitalpha.richtext.widget.tools.manager.LinkManagerImpl;
+import org.polarsys.kitalpha.richtext.widget.tools.utils.Constants;
+
+/**
+ * Using SAX parser replace invalid link with another element in description
+ */
+public abstract class AbstractReplaceInvalidHyperLinkInDescription {
+
+  public boolean updateDescription(List<EObject> modelElements, String linkId, String label) {
+
+
+    EObject object = modelElements.get(0);
+    if (object instanceof CapellaElement || object instanceof DRepresentationDescriptor) {
+      LinkManager linkManager = new LinkManagerImpl() {
+        @Override
+        public List<String> getAllLinkLabels() {
+          return getLinkLabels();
+        }
+      };
+      MDELinkDialog dialog = new MDELinkDialog(Display.getCurrent().getActiveShell(), (EObject) object, label,
```

Comment:
A small change to make this class testable would be to depend on an interface instead of MDELinkDialog. Then a headless resolver can be provided to unit test the description text replacement for example.